backends-velox/src/main/scala/io/glutenproject/execution/ColumnarToVeloxColumnarExec.scala

```diff
@@ -0,0 +1,158 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.glutenproject.execution
+
+import io.glutenproject.backendsapi.velox.ValidatorApiImpl
+import io.glutenproject.columnarbatch.ColumnarBatches
+import io.glutenproject.exec.Runtimes
+import io.glutenproject.memory.arrowalloc.ArrowBufferAllocators
+import io.glutenproject.memory.nmm.NativeMemoryManagers
+import io.glutenproject.utils.{ArrowAbiUtil, Iterators}
+import io.glutenproject.vectorized.VanillaColumnarToNativeColumnarJniWrapper
+
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.execution.SparkPlan
+import org.apache.spark.sql.execution.arrow.ArrowColumnarBatchConverter
+import org.apache.spark.sql.execution.metric.SQLMetric
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.utils.SparkArrowUtil
+import org.apache.spark.sql.vectorized.ColumnarBatch
+import org.apache.spark.util.TaskResources
+
+import org.apache.arrow.c.{ArrowArray, ArrowSchema, Data}
+
+case class ColumnarToVeloxColumnarExec(child: SparkPlan) extends ColumnarToColumnarExecBase(child) {
+
+  override def doExecuteColumnarInternal(): RDD[ColumnarBatch] = {
+    new ValidatorApiImpl().doSchemaValidate(schema).foreach {
+      reason =>
+        throw new UnsupportedOperationException(
+          s"Input schema contains unsupported type when performing columnar" +
+            s" to columnar for $schema " + s"due to $reason")
+    }
```

Comment:
It looks a little bit weird to do validation during execution. Would you like to share the reason doing this?

Reply:
Yea, I follow `RowToVeloxColumnarExec` to do the validation here, you're right, it looks weird to do so during execution, I also have the same doubt when seeing `RowToVeloxColumnarExec`...

Let me fix it by overriding `doValidateInternal` instead.